src/modules/model.js

```diff
@@ -66,6 +74,21 @@ class Model {
     });
   }
 
+  // eslint-disable-next-line no-underscore-dangle
+  _generateId(data, providedId) {
+    if ((providedId === null) === (this.primaryKeys === null)) {
```

Comment:
```suggestion
    if ((typeof providedId !== 'string') === (!Array.isArray(this.primaryKeys))) {
```

plus add more asserts?